api_tests/test_functional.py

```diff
@@ -32,3 +39,51 @@ def test_support_selective_serialization():
     Only serializable globals are returned; others are simply omitted.
     """
     assert {"out2": "hi"} == call_api_success("out1 = object(); out2 = 'hi'", {})
+
+
+def test_support_exception_traceback_stderr():
+    """
+    Exceptions from submitted code are serialized and returned, along with stderr.
+
+    Globals aren't updated when there's an exception, even via mutation.
```

Comment:
Even as I write this, I'm not clear on the point of this test in the the API tests. I could imagine a unit test, but
isn't the purpose of these tests to ensure you have a secure environment?
```suggestion
    Globals aren't updated when there's an exception, even via mutation. This is not for reasons of
    security, but simply to ensure compatibility with the current codejail implementation. See
    code comments for details.
```

Reply:
That was part of the ticket that's behind this work: https://github.com/edx/edx-arch-experiments/issues/896 -- functionality and security testing.

Everything in `test_functional.py` is just for basic functionality testing -- ensuring that we didn't break the execution environment by overdoing confinement, and that we conform to the API that edxapp expects.